### ISSUE TITLE
Can't connect to Mongo using MongoClient

### ISSUE THREAD

**Issue by tsirolnik** [mongodb-js/mongodb-core#265]
Trying to do the following using mongodb 3.0 

```js
let MongoClient = require('mongodb').MongoClient;
let mongoURI = `<connection string>`;
MongoClient.connect(mongoURI, (error, db) => {
    if (error) {
        console.log('MongoDB connect', error);
        return;
    }
    //...
});
```

I receive this error - 
```
MongoDB connect TypeError: Cannot read property 'options' of undefined
    at makeSSLConnection (/xxx/node_modules/mongodb-core/lib/connection/connection.js:538:25)
    at doConnect (/xxx/node_modules/mongodb-core/lib/connection/connection.js:610:7)
    at Connection.connect (/xxx/node_modules/mongodb-core/lib/connection/connection.js:641:10)
    at Pool.connect (/xxx/node_modules/mongodb-core/lib/connection/pool.js:774:16)
    at Server.connect (/xxx/node_modules/mongodb-core/lib/topologies/server.js:571:25)
    at Timeout._onTimeout (/xxx/node_modules/mongodb-core/lib/topologies/replset.js:414:14)
    at ontimeout (timers.js:469:11)
    at tryOnTimeout (timers.js:304:5)
    at Timer.listOnTimeout (timers.js:264:5)
```

Digging into to line 538 @ connection.js shows this - 
```js
 // Merge in options
  merge(sslOptions, this.options);
  merge(sslOptions, _options);
```

Changing ``this.options`` to ``self.options`` seems to solve the problem. 

Edit: This seems to be a bug that came with version 3.0.0 on npm as I'm using version 2.2.33 on a different system and it works just fine

**Comment by mbroadst** [comment]
@tsirolnik Thanks for catching this! I'm putting together a patch release today for a few other minor issues discovered. It should also be mentioned that the `connect` signature has changed (if you're upgrading to 3.x) from `MongoClient.connect(mongoURI, (error, db) =>` to `MongoClient.connect(mongoURI, (error, client) =>`. Please take a look at our [migration guide](https://github.com/mongodb/node-mongodb-native/blob/3.0.0/CHANGES_3.0.0.md) for other interesting changes.

**Comment by mbroadst** [comment]
fixed in 06b938810ef11fb210e3221b3bbfd9925b6ddc2d

**Comment by tsirolnik** [comment]
np, thanks for the quick fix!
